### PR TITLE
Fix/issue 918 - Status Site - Feature flag to remove redundant task status tabs

### PR DESCRIPTION
**Pull request by aneenajohn**
### Description: 

There are two ways to filter the tasks based on task status on [Tasks Page](https://status.realdevsquad.com/tasks) as shown below.
- **Filter 1** - Drop down filter
- **Filter 2** - Tabs to filter out the task as marked in the below image -- Need to remove this

image

### Anything you would like to inform the reviewer about:
- Have removed the filter by tabs component 
- Have fixed the search button handler and search input handling failing test cases
- Have updated the test cases related to Tabs Comp to test only when dev is true

### Dev Tested:
- [x] Yes
### Is Under Feature Flag:
- [x] Yes
- [ ] No
image

#### Large Screens Before:
image

#### Small Screens Before:
image

##### After:
https://github.com/Real-Dev-Squad/website-status/assets/75213145/c0ba702a-9b9c-4be0-abe2-8fa8781cdadc

### Follow-up Issues (if any)
NA

### CODE REVIEW

**Comment by sahsisunny** [comment]
Please add test coverage report or screenshot.

**Comment by aneenajohn** [comment]
> Please add test coverage report or screenshot.

Please find the complete report below:

- [Test report on working branch](https://docs.google.com/document/d/198jqsvHOw8Aypy9NZeZKYaYvurLTfbAxgON9Az75u2g/edit?usp=sharing)
- [Test report on development branch](https://docs.google.com/document/d/1_wSpTxdsdy169SkI6AIxWYDEUCBpzmcU4uJO5YQ2gCM/edit#heading=h.7husqmtv2ro4)

**Review by iamitprakash — CHANGES_REQUESTED** [review]
test coverage stats

**Comment by sahsisunny** [comment]
Can you add a feature flag to show the secondary dropdown? Because we did not discuss it with @ankurnarkhede and @iamitprakash (Tejas) to remove it.

**Comment by aneenajohn** [comment]
> test coverage stats

image

File: src/components/tasks/TasksContent.tsx [hunk 1]
```diff
@@ -143,40 +144,48 @@ export const TasksContent = ({ dev }: { dev?: boolean }) => {
...
+                                        queryAssignees,
+                                        queryTitle
+                                    );
+                                }
+                            }}
+                            options={taskSelectOptions}
+                        />
+                    </div>
+                </>
+            ) : null}
```
Comment (shubhamsigdar1) [inline src/components/tasks/TasksContent.tsx:188]: Donot understand when feature flag then you show tab??

Comment (aneenajohn) [reply src/components/tasks/TasksContent.tsx:188]: Thanks for point that, Have updated this we don't have to show this tab when feature flag is on

**Comment by aneenajohn** [comment]
@shubhamsigdar1 / @sahsisunny / @iamitprakash Please review

**Review by iamitprakash — CHANGES_REQUESTED** [review]
please add proper PR desc

**Review by shubhamsigdar1 — COMMENTED** [review]
Want to know you taken latest pull from develop because my pr got merged https://github.com/Real-Dev-Squad/website-status/pull/960

**Comment by aneenajohn** [comment]
> Want to know you taken latest pull from develop because my pr got merged #960

@shubhamsigdar1, Yes latest pull from develop is pulled into this working branch

**Comment by aneenajohn** [comment]
> please add proper PR desc

Updated now, please check

File: src/components/tasks/TasksContent.tsx [hunk 2]
```diff
@@ -143,40 +144,48 @@ export const TasksContent = ({ dev }: { dev?: boolean }) => {
...
```
Comment (Pratiyushkumar) [inline src/components/tasks/TasksContent.tsx:188]: why we are using `null` and not the empty string or react fragments?

File: src/components/tasks/TasksContent.tsx [hunk 3]
```diff
@@ -143,40 +144,48 @@ export const TasksContent = ({ dev }: { dev?: boolean }) => {
...
-                                queryAssignees,
-                                queryTitle
-                            );
-                        }
-                    }}
-                    options={taskSelectOptions}
-                />
-            </div>
+            {dev !== true ? (
+                <>
```
Comment (Pratiyushkumar) [inline src/components/tasks/TasksContent.tsx:148]: remove this fragment if not needed here

File: src/constants/task.ts
Comment (Pratiyushkumar) [inline src/constants/task.ts:1]: why is this needed? dont we have a constant file created for it

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx [hunk 4]
```diff
@@ -109,22 +87,20 @@ describe('tasks content', () => {
...
-        await screen.findByTestId('tabs');
 
         const task = await findByText(
             'Design and develop an online booking system'
         );
         expect(task).toBeInTheDocument();
     });
 
-    test('display tasks when dev is true', async () => {
+    test('display tasks', async () => {
```
Comment (Pratiyushkumar) [inline __tests__/Unit/Components/Tasks/TasksContent.test.tsx:97]: can you please write proper test case message, 'display tasks' doesn't give me sense

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx [hunk 5]
```diff
@@ -332,14 +235,12 @@ describe('tasks content', () => {
...
-
-        await screen.findByTestId('tabs');
-        const searchInput = screen.getByTestId('search-input');
+        const searchInput = await screen.findByTestId('search-input');
         fireEvent.change(searchInput, { target: { value: 'test' } });
         expect(searchInput).toHaveValue('test');
     });
 
-    test('Query param should be changed when tab is selected', async () => {
+    test('Query param should be changed when tab is selected when', async () => {
```
Comment (Pratiyushkumar) [inline __tests__/Unit/Components/Tasks/TasksContent.test.tsx:243]: what is the change here??

File: src/styles/tasks.module.scss [hunk 6]
```diff
@@ -1,13 +1,27 @@
...
+    width: 64%;
+    @media (max-width: 1280px) {
+        width: 85%;
+    }
+    @media (max-width: 768px) {
+        width: 90%;
+    }
+    @media (max-width: 600px) {
+        width: 100%;
+    }
```
Comment (Pratiyushkumar) [inline src/styles/tasks.module.scss:14]: why cant we follow the proper rule for media queries, why cant we separate out the media query instead of crowding everything into one place?

**Review by Pratiyushkumar — DISMISSED** [review]
Also, please actively get this PR merged as this PR is a blocker for all the tasks regarding task card enforcement.

File: src/components/tasks/TasksContent.tsx (hunk 3 above)
Comment (aneenajohn) [reply src/components/tasks/TasksContent.tsx:148]: This fragment is needed here as we have to wrap to <div> together

File: src/components/tasks/TasksContent.tsx (hunk 2 above)
Comment (aneenajohn) [reply src/components/tasks/TasksContent.tsx]: This is updated now, Thanks for pointing that out!

File: src/constants/task.ts
Comment (aneenajohn) [reply src/constants/task.ts:1]: I believe this is taken during the pull from development branch while resolving conflicts

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx (hunk 5 above)
Comment (aneenajohn) [reply __tests__/Unit/Components/Tasks/TasksContent.test.tsx]: This is a typo, has been removed now!

File: src/constants/task.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    AVAILABLE: [],
+    UNASSIGNED: [],
+    NEEDS_REVIEW: [],
+    IN_REVIEW: [],
+    VERIFIED: [],
+    MERGED: [],
+    COMPLETED: [],
+    OVERDUE: [],
+    DONE: [],
+};
```
Comment (shubhamsigdar1) [inline src/constants/task.ts:14]: This above is already in https://github.com/Real-Dev-Squad/website-status/blob/develop/src/constants/tasks.ts#L1

Comment (aneenajohn) [reply src/constants/task.ts]: @shubhamsigdar1, Yes,This is removed now as this was added to resolve conflicts in the early stages

File: src/constants/task.ts
Comment (aneenajohn) [reply src/constants/task.ts:1]: Yes we have, Removed now as this was added to resolve conflicts in the early stages

**Review by iamitprakash — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]
